src/dft/src/stitch/ScanStitch.hh

```diff
@@ -48,18 +53,45 @@ class ScanStitch
 
   // Stitch all the cells inside each one of the scan chains together.
   void Stitch(const std::vector<std::unique_ptr<ScanChain>>& scan_chains);
-  void Stitch(odb::dbBlock* block, const ScanChain& scan_chain);
+  void Stitch(odb::dbBlock* block,
+              const ScanChain& scan_chain,
+              const ScanDriver& scan_enable);
 
  private:
-  void FindOrCreateScanEnable(odb::dbBlock* block);
+  ScanDriver FindOrCreateScanEnable(odb::dbBlock* block);
   ScanDriver FindOrCreateScanIn(odb::dbBlock* block);
   ScanLoad FindOrCreateScanOut(odb::dbBlock* block,
                                const ScanDriver& cell_scan_out);
 
+  // Typesafe function to create Ports for the scan chains.
+  template <typename Port>
+  Port CreateNewPort(odb::dbBlock* block, std::string_view name_pattern)
+  {
+    for (uint64_t port_number = 1;; ++port_number) {
```

Comment:
Are you expecting more the 2^32 ports?  From https://google.github.io/styleguide/cppguide.html#Integer_Types

```
We use int very often, for integers we know are not going to be too big, e.g., loop counters. Use plain old int for such things.```

Reply:
Good catch, thanks!